Source/ASButtonNode.mm

```diff
@@ -553,6 +541,118 @@ - (UIAccessibilityTraits)defaultAccessibilityTraits
                       : (UIAccessibilityTraitButton | UIAccessibilityTraitNotEnabled);
 }
 
+#pragma mark - Layout
+
+#if !YOGA
+- (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
+{
+    UIEdgeInsets contentEdgeInsets;
+    ASButtonNodeImageAlignment imageAlignment;
+    ASLayoutSpec *spec;
+    ASStackLayoutSpec *stack = [[ASStackLayoutSpec alloc] init];
+    {
+        ASLockScopeSelf();
+        stack.direction = _laysOutHorizontally ? ASStackLayoutDirectionHorizontal : ASStackLayoutDirectionVertical;
+        stack.spacing = _contentSpacing;
+        stack.horizontalAlignment = _contentHorizontalAlignment;
+        stack.verticalAlignment = _contentVerticalAlignment;
+        
+        contentEdgeInsets = _contentEdgeInsets;
+        imageAlignment = _imageAlignment;
+    }
+    
+    NSMutableArray *children = [[NSMutableArray alloc] initWithCapacity:2];
+    if (_imageNode.image) {
+        [children addObject:_imageNode];
+    }
+    
+    if (_titleNode.attributedText.length > 0) {
+        if (imageAlignment == ASButtonNodeImageAlignmentBeginning) {
+            [children addObject:_titleNode];
+        } else {
+            [children insertObject:_titleNode atIndex:0];
+        }
+    }
+    
+    stack.children = children;
+    
+    spec = stack;
+    
+    if (UIEdgeInsetsEqualToEdgeInsets(UIEdgeInsetsZero, contentEdgeInsets) == NO) {
+        spec = [ASInsetLayoutSpec insetLayoutSpecWithInsets:contentEdgeInsets child:spec];
+    }
+    
+    if (_backgroundImageNode.image) {
+        spec = [ASBackgroundLayoutSpec backgroundLayoutSpecWithChild:spec background:_backgroundImageNode];
+    }
+    
+    return spec;
+}
+#endif
+
+- (void)updateYogaLayoutIfNeeded
+{
+#if YOGA
+  NSMutableArray<ASDisplayNode *> *children = [[NSMutableArray alloc] initWithCapacity:2];
```

Comment:
Could this be pulled out into a category or subclass?

Reply:
Done